os/kernel/log_dump/log_dump.c

```diff
@@ -125,33 +142,39 @@ static void log_dump_mem_check(size_t max_size)
 	}
 }
 
-int log_dump_save(char ch)
+int log_dump_tobuffer(char ch, size_t *free_size)
 {
 	/* need to check if the current chunks size is over max_log_size or greater than x% of free heap */
 
-	size_t free_size;
+	size_t max_size;
 #ifdef CONFIG_DEBUG_MM_HEAPINFO
-	free_size = mm_get_heap_free_size(void);
+	*free_size = mm_get_heap_free_size();
 #else
 	struct mallinfo mem;
 #ifdef CONFIG_CAN_PASS_STRUCTS
 	mem = kmm_mallinfo();
 #else
 	(void)kmm_mallinfo(&mem);
 #endif
-	free_size = mem.fordblks;
+	*free_size = mem.fordblks;
 #endif
-	size_t percent_free_size = (free_size * CONFIG_LOG_DUMP_MAX_FREE_HEAP) / 100;
-	size_t max_size = MIN(percent_free_size, CONFIG_LOG_DUMP_MAX_SIZE);             /* setting the upper limit */
-	max_size = MAX(max_size, CONFIG_LOG_DUMP_MIN_SIZE);                             /* checking the lower limit */
-
-	struct log_dump_chunk_s *log_dump_tail = (struct log_dump_chunk_s *)sq_tail(&log_dump_chunks);
+	size_t percent_free_size = (*free_size * CONFIG_LOG_DUMP_MAX_FREE_HEAP) / 100;
+	size_t localbytes = 0;
+
+	if(percent_free_size < CONFIG_LOG_DUMP_MAX_SIZE) {
+		max_size = LOG_CHUNK_SIZE;
+	} else {
+		max_size = MIN(percent_free_size, CONFIG_LOG_DUMP_MAX_SIZE);	/* setting the upper limit */
+		max_size = MAX(max_size, CONFIG_LOG_DUMP_MIN_SIZE);		/* checking the lower limit */
+	}
 
 	if (curbytes == CONFIG_LOG_DUMP_CHUNK_SIZE) { /* last chunk is full */
 
 		if (log_dump_size + LOG_CHUNK_SIZE < max_size) {	
 			/* memory available, so fetch a new buffer */
 			struct log_dump_chunk_s *node = (struct log_dump_chunk_s *)kmm_malloc(LOG_CHUNK_SIZE);
+			localbytes = compressBound(LOG_CHUNK_SIZE);
+			lldbg("sangam dbg : for 4k size = %d\n", localbytes);
```

Comment:
Could you remove this debug line please?

Reply:
done